src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -506,17 +507,28 @@ internal static string GetPipeNameForPathOpt(string compilerExeDirectory)
             return $"{userName}.{isAdmin}.{basePipeName}";
         }
 
-        internal static string GetBasePipeName(string compilerExeDirectory)
+        internal static string GetBasePipeName(string compilerExeDirectory, string sharedCompilationId = null)
         {
             // Normalize away trailing slashes.  File APIs include / exclude this with no 
             // discernable pattern.  Easiest to normalize it here vs. auditing every caller
             // of this method.
             compilerExeDirectory = compilerExeDirectory.TrimEnd(Path.DirectorySeparatorChar);
 
+            // Store refined compilerExeDirectory value as the basis for a raw pipe name to be hashed.
+            string rawPipeName = compilerExeDirectory;
```

Comment:
That is partially correct; the **base** pipe name is a combination of `compilerExeDirectory + '\0' + sharedCompilationId`, with the null terminator acting as a delimiter (see [explanation](https://github.com/dotnet/roslyn/pull/20374#discussion_r126472005) for delimiter pattern below). The base pipe name is then hashed using the same convention as it was before this change, and username and elevation are then also included as ingredients - as before.

Per an email thread, to which all participants of this PR were added, with last reply sent on 13-June-2017, it appeared to me that the consensus was that the sharedCompilationId value would be an ingredient of the fully constructed pipe name; last response from @jaredpar :

> Yes I agree the pipe name is essentially a combination of three values: user defined value, elevation state and path to VBCSCompiler.
>
>	Change the meaning of the value of the /shared switch.
>
>I think this is the best option.  It would mean the MSBuild property essentially directly maps to the equivalent command line switch.  That is the standard pattern in that pipeline.

With no dissent following this; my assumption was that this was the path to take.